lib/runner/variant-reassignment.js

```diff
@@ -17,52 +16,148 @@ export default class VariantReassignment {
     this.transactionService = new TransactionService(logger, client)
   }
 
+  /**
+   * Take a list of product drafts and existing products matched by sku
+   *  - for every productDraft check if reassignment is needed
+   *  - if yes, create and process actions which will move variants across products
+   * @param productDrafts List of productDrafts
+   * @param existingProductProjections List of existing products matching by SKU
+   * @returns {Promise.<*>}
+   */
   async execute (productDrafts, existingProductProjections) {
-    this._processUnfinishedTransactions()
+    let products
+
+    try {
+      if (this.firstRun)
+        await this._processUnfinishedTransactions()
+    } catch (e) {
+      return this._error('Could not process unfinished transactions', e)
+    }
+    this.firstRun = false
 
-    const products
-      = await this.productService.fetchProductsFromProductProjections(existingProductProjections)
+    try {
+      products
+        = await this.productService.fetchProductsFromProductProjections(existingProductProjections)
+    } catch (e) {
+      return this._error('Error while fetching products for reassignment', e)
+    }
 
     const productDraftsForReassignment
       = this._selectProductDraftsForReassignment(productDrafts, products)
 
-    if (productDraftsForReassignment.length)
-      for (const productDraft of productDraftsForReassignment)
+    this.logger.debug(
+      'Filtered %d productDrafts for reassignment',
+      productDraftsForReassignment.length
+    )
+
+    for (const productDraft of productDraftsForReassignment)
+      try {
         await this._processProductDraft(productDraft, products)
+      } catch (e) {
+        this.logger.error(
+          'Error while processing productDraft %j, retrying.',
+          productDraft.name, e.stack
+        )
+        await this._handleProcessingError(productDraft, products)
+      } finally {
+        this.logger.debug(
+          'Finished processing of productDraft with name %j',
+          productDraft.name
+        )
+      }
+
+    return Promise.resolve()
```

Comment:
This is because of a linter error:
```
lib/runner/variant-reassignment.js
  30:9  error  Expected to return a value at the end of async method 'execute'  consistent-return
```

Sure it can be disabled but I think it can be get handy some day